src/CommunityToolkit.Maui.MediaElement/Views/MediaManager.android.cs

```diff
@@ -0,0 +1,423 @@
+			_ => MediaElementState.None,
+		};
+
+		mediaElement.CurrentStateChanged(newState);
+
+		if (playbackState is IPlayer.StateReady)
+		{
+			mediaElement.Duration = TimeSpan.FromMilliseconds(player.Duration < 0 ? 0 : player.Duration);
+		}
+	}
+
+	/// <summary>
+	/// Occurs when ExoPlayer changes the playback state.
+	/// </summary>
+	/// <paramref name="playbackState">The state that the player has transitioned to.</paramref>
+	/// <remarks>
+	/// This is part of the <see cref="IPlayer.IListener"/> implementation.
+	/// While this method does not seem to have any references, it's invoked at runtime.
+	/// </remarks>
+	public void OnPlaybackStateChanged(int playbackState)
+	{
+		if (mediaElement.Source is null)
+		{
+			return;
+		}
+
+		MediaElementState newState = mediaElement.CurrentState;
+
+		switch (playbackState)
+		{
+			case IPlayer.StateBuffering:
+				newState = MediaElementState.Buffering;
+				break;
+			case IPlayer.StateEnded:
+				mediaElement.MediaEnded();
+				break;
+		}
+
+		mediaElement.CurrentStateChanged(newState);
+	}
+
+	/// <summary>
+	/// Occurs when ExoPlayer encounters an error.
+	/// </summary>
+	/// <paramref name="error">An instance of <seealso cref="PlaybackException"/> containing details of the error.</paramref>
+	/// <remarks>
+	/// This is part of the <see cref="IPlayer.IListener"/> implementation.
+	/// While this method does not seem to have any references, it's invoked at runtime.
+	/// </remarks>
+	public void OnPlayerError(PlaybackException? error)
+	{
+		if (mediaElement is null)
+		{
+			return;
+		}
+
+		var errorMessage = string.Empty;
+		var errorCode = string.Empty;
+		var errorCodeName = string.Empty;
+
+		if (!string.IsNullOrWhiteSpace(error?.LocalizedMessage))
+		{
+			errorMessage = $"Error message: {error.LocalizedMessage}";
+		}
+
+		if (error?.ErrorCode is not null)
+		{
+			errorCode = $"Error code: {error?.ErrorCode}";
+		}
+
+		if (!string.IsNullOrWhiteSpace(error?.ErrorCodeName))
+		{
+			errorCode = $"Error codename: {error?.ErrorCodeName}";
+		}
+
+		var message = string.Join(", ", new[] { errorCodeName, errorCode, errorMessage }.Where(s => !string.IsNullOrEmpty(s)));
+
+		mediaElement.MediaFailed(new MediaFailedEventArgs(message));
+
+		Logger?.LogError("{logMessage}", message);
+	}
+
+	/// <summary>
+	/// Invoked when a seek operation has been processed.
+	/// </summary>
+	/// <remarks>
+	/// This is part of the <see cref="IPlayer.IListener"/> implementation.
+	/// While this method does not seem to have any references, it's invoked at runtime.
+	/// </remarks>
+	public void OnSeekProcessed()
+	{
+		mediaElement?.SeekCompleted();
+	}
+
+	/// <summary>
+	/// Occurs when ExoPlayer changes volume.
+	/// </summary>
+	/// <param name="volume">The new value for volume.</param>
+	/// <remarks>
+	/// This is part of the <see cref="IPlayer.IListener"/> implementation.
+	/// While this method does not seem to have any references, it's invoked at runtime.
```

Comment:
So the reason I did this is because these actually do something so they're more discoverable. The ones that don't do anything but we need to have them because else Android will break are in the region below. I though this made most sense so that people don't spend time looking for the methods that actually have functionality and I added a comment to make it more clear 😅 I think I'll leave it like this for now, if it gets in the way or anything let's change it then.